Anagram.java

```diff
@@ -24,26 +26,79 @@ public static void main(String args[]) {
 			if (!pass) break;
 		}
 		System.out.println(pass ? "test passed" : "test Failed");
-	}  
+	}
 
 	// Returns true if the two given strings are anagrams, false otherwise.
 	public static boolean isAnagram(String str1, String str2) {
-		// Replace the following statement with your code
-		return false;
-	}
+		String main_str = preProcess(str1);
+		String secondary_str = preProcess(str2);
+
+		for (int i = 0; i < main_str.length(); i++){
+			for (int j = 0; j < secondary_str.length(); j++){
+				if (main_str.charAt(i) == secondary_str.charAt(j)){
+
+					secondary_str = secondary_str.replaceFirst(Character.toString(secondary_str.charAt(j)), "");
```

Comment:
Both string.replaceFirst() and Character.toString() wasn't taught in recitation or lecture, therefore it is hw guidelines violation. -4